cocotb/regression.py

```diff
@@ -406,9 +400,16 @@ def _result_was():
 
         return result_pass, sim_failed
 
-    def execute(self):
-        self._running_test = cocotb.regression_manager.next_test()
-        if self._running_test:
+    def execute(self) -> None:
+        while True:
+            self._test = self.next_test()
+            if not self._test:
+                return self.tear_down()
+
+            self._running_test = self._init_test(self._test)
+            if not self._running_test:
+                continue
+
```

Comment:
Can you unindent everything below this line to be out of the loop?

Reply:
This style looks a bit odd, but it will make sense in following commits and I didn't want to diff it back again. The unconditioned break below will be conditioned on whether the test finished after the call to `scheduler.add`. In the case of tests which finish immediately like:

```python
@cocotb.test()
async def returns_immediately(dut):
    return 1
```

Reply:
Can you perhaps then instead break out a helper `_execute_one(self) -> bool`, so that this instead becomes:

```
while True:
    self._test = self.next_test()
    returned_immediately = self._execute_one()
    if not returned_immediately:
        break